src/core/lib/event_engine/posix_engine/posix_engine.cc

```diff
@@ -332,7 +332,7 @@ PosixEventEngine::PosixEventEngine(PosixEventPoller* poller)
     : connection_shards_(std::max(2 * gpr_cpu_num_cores(), 1u)),
       executor_(std::make_shared<ThreadPool>()),
       timer_manager_(executor_) {
-  if (UseEventEngineClient()) {
+  if (UseEventEngineClient() || UseEventEngineServer()) {
```

Comment:
This would read better if we had a `bool NeedPosixEngine() { return UseClient() || UseServer(); }`
so you can write `!NeedPosixEngine()` below too

Reply:
Good point. Fixed.